src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+
+* `component {Component}`
+* `gradeName {String}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied component has the supplied `gradeName` as a parent grade. This is a rapid check which uses the framework's internal data structures for the lookup.
+
+
+## Basic implementation atoms
+
+### fluid.identity(arg)
+
+* `arg {Any}`
+* Returns: `{Any}`
+
+### fluid.notImplemented()
+
+A function which raises a failure if executed. This function name can also be used as a placeholder in [Invokers](Invokers.md) and
+namespaced [Listener](InfusionEventSystem.md) to create abstract (noninstantiable) [Component Grades](ComponentGrades.md).
+
+### fluid.each(source, func)
+
+* `source {Array|Object}` A container to be iterated over (may be null)
+* `func {Function} (element {Any}, index {Number|String}) → Any`. This function will be invoked once on each member of `source` and will receive the member and its key (either an integer for an array source,
+or a string for an object source) 
+* Returns: none
+
+Iterates over a supplied array or hash, invoking a function once for each element. Similar to [`jQuery.each`](http://api.jquery.com/jquery.each/) only the
+arguments to `func` are the right way round and the function does not explode on nonvalues. `fluid.each` on `null` or `undefined` is a no-op.
+
+### fluid.invokeLater(func) 
+
+A standard utility to schedule the invocation of a function after the current function call stack returns. On browsers this defaults to `setTimeout(func, 1)` but in
+other environments can be customised - e.g. to `process.nextTick` in node.js. In future, this could be optimised in the browser to not dispatch into the event queue
+
+* `func {Function}` A function accepting zero arguments which is to be invoked later
+* Returns: none
+
+### fluid.allocateGuid()
+
+Allocate a string value that will be unique within this Infusion instance (frame or process), and
+globally unique with high probability (50% chance of collision after a million trials). This has the structure `prefix-id` where
+`prefix` is the value of `fluid.fluidInstance` identifying this Infusion instance, and `id` is a increasing integer initialised at
+1 when this infusion instance starts up.
+
+* Returns: `{String}` A string of about a dozen characters that has made moderate efforts to be globally unique.
+
+
+## Creation, copying and destroying
+
+### fluid.freshContainer(tocopy)
+
+* `tocopy {Object|Array}`
+* Returns: `{Object|Array}`
+
+Return an empty container as the same type as the argument (either an array or hash). To be wise, the supplied argument should 
+pass the `fluid.isPlainObject` test.
+
+### fluid.copy(tocopy)
+
+* `tocopy {Any}`
+* Returns: `{Any}`
+
+Performs a deep copy (clone) of its argument. This will guard against cloning a circular object by terminating if it reaches a path depth
+greater than `fluid.strategyRecursionBailout`. `fluid.copy` will not copy an object which passes `fluid.isUncopyable`.
```

Comment:
We might want to mention what the default value of fluid.strategyRecursionBailout is